src/components/movieCard/MovieCard.tsx

```diff
@@ -0,0 +1,29 @@
+import { StyleSheet, View, Text, TouchableOpacity, Image } from 'react-native';
+
+import styles from '@components/movieCard/style';
+import { memo } from 'react';
+
+function MovieCard({
+  movie,
+  onPress,
+  containerStyle,
+  headingStyle,
+  imageStyle,
+}: MovieCardProps) {
+  return (
+    <TouchableOpacity
+      style={styles.container}
+      onPress={onPress}>
+      <Image
+        source={{ uri: movie.posterUrl }}
+        style={styles.imageStyle}
+        resizeMode="cover"
+      />
+        <Text numberOfLines={1} ellipsizeMode='tail' style={StyleSheet.compose(styles.headingStyle, headingStyle)}>
```

Comment:
why using `Stylesheet.compose`??

Reply:
`Stylesheet.compose` is more efficient compared to the regular `array` method. Every time we use the regular `array` method to pass `styles` to components it creates a new array even if one of the `style` object is `null` or `falsy` but if we use the `Stylesheet.compose` method it will directly return the non-`falsy` or non-`null` style object without creating a new style object.